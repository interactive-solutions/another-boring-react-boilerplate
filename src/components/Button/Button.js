// @flow
import React from 'react';

import './Button.scss';

type Props = {
  text?: string,
  alertText?: string,
};

const Button = (props: Props) => (
  <button
    styleName="button"
    onClick={() => {
      alert(props.alertText); // eslint-disable-line
    }}
  >
    {props.text}
  </button>
);

Button.defaultProps = {
  text: 'Button',
  alertText: 'I was clicked!',
};

export default Button;
