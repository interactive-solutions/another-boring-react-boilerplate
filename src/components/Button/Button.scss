.button {
  background: $primaryColor;
  color: #fff;
  border: none;
  border-radius: $defaultBorderRadius;
  padding: 10px 25px;
  @include font-size(14);
  cursor: pointer;
  transition: all, 0.2s;

  &:hover {
    background: darken($primaryColor, 3%);
    box-shadow: 0 0 5px rgba(0, 0, 0, 0.25);
  }

  &:focus {
    outline: 0;
  }

  &:active {
    background: darken($primaryColor, 10%);
    box-shadow: none;
  }
}
