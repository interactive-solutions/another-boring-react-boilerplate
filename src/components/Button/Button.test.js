import React from 'react';
import Button from './Button';

describe('Button', () => {
  const button = shallow(<Button text="I am a button" />);

  it('should render correctly', () => {
    expect(button).toMatchSnapshot();
  });

  it('should have class .button', () => {
    expect(button.hasClass('button')).toBeTruthy();
  });

  it('should display passed text prop', () => {
    expect(button.text()).toBe('I am a button');
  });

  it('should handle the click event and alert the default text', () => {
    window.alert = jest.fn();
    button.simulate('click');
    expect(window.alert).toHaveBeenCalledWith('I was clicked!');
  });

  it('should display custom alert text if alertText prop is provided', () => {
    button.setProps({ alertText: 'Hi!' });
    button.simulate('click');
    expect(window.alert).toHaveBeenCalledWith('Hi!');
  });
});
