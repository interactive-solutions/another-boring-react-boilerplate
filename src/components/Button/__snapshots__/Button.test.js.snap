// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  I am a button
</button>
`;
